docs/removal-behavior.md

```diff
@@ -0,0 +1,56 @@
+# Removal Behavior for interfaces
+
+## `unusedinterfaces` optimization
+
+The `unusedinterfaces` optimization controls whether or not classes have an annotation that denotes whether a class implements an interface. When the optimization is off, the linker will not remove the annotations regardless of the visibility of the interface (even private interface implementations will be kept). When the optimization is on and the linker can provably determine that an interface will not be used on a type, the annotation will be removed. In order to know whether it is safe to remove an interface implementation, it is necessary to have a "global" view of the code. In other words, if an assembly is passed without selecting `link` for the `action` option for the linker, all classes that implement interfaces from that assembly will keep those interface implementation annotations. For example, if you implement `System.IFormattable` from the `System.Runtime` assembly, but pass the assembly with `--action copy System.Runtime`, the interface implementation will be kept even if your code doesn't use it.
+
+## Static abstract interface methods
+
+The linker's behavior for methods declared on interfaces as `static abstract` like below are defined in the following cases:
```

Comment:
I'd follow a consistent style here of

`[rule]`
`[example]`

Instead of something more like prose. Basically, this is code, but in English.

I think the phrasing could also be more specific in what's rooted and what's not.

Something like:

> On a direct call to a static method which implements a static interface method, only the body is rooted, not its associated MethodImpl. Similarly, the interface method which it implements is not rooted.

Reply:
Overall, if we imagine we had a complete spec here, I expect it would look something like:

`[IL instruction]`
`[description of things it roots]`
`[example]`